octopus_deploy/tests/test_unit.py

```diff
@@ -869,3 +869,21 @@ def test_server_node_endpoint_failed(get_current_datetime, dd_run_check, aggrega
             'server_node_name:octopus-i8932-79236734bc234-09h234n',
         ],
     )
+
+
+@pytest.mark.parametrize(
+    ('expected_events', 'events_enabled'),
+    [pytest.param([], False, id='events disabled'), pytest.param(ALL_EVENTS, True, id='events enabled')],
+)
+@pytest.mark.usefixtures('mock_http_get')
+@mock.patch("datadog_checks.octopus_deploy.check.get_current_datetime")
+def test_events(get_current_datetime, dd_run_check, aggregator, expected_events, events_enabled):
+    instance = {'octopus_endpoint': 'http://localhost:80'}
+    instance['collect_events'] = events_enabled
+    check = OctopusDeployCheck('octopus_deploy', {}, [instance])
+    get_current_datetime.return_value = MOCKED_TIME1
+    dd_run_check(check)
+    get_current_datetime.return_value = MOCKED_TIME2
+    dd_run_check(check)
```

Comment:
Gave the LGTM, but just out of curiosity, I wanted to ask why you needed to run the check twice with different times to report the events correctly.

Reply:
Thanks- the first time the check is run, the `from` and `to` request parameters are the same, set here: https://github.com/DataDog/integrations-core/blob/sarah/add-octopus-integration/octopus_deploy/datadog_checks/octopus_deploy/check.py#L39-L44 and used here: https://github.com/DataDog/integrations-core/pull/19176/files#diff-55d4c28ac5fa39555c3ea453a3b7181421b4c62576a462a4989b24ba69888308R313-R315. Every other check run they are ~15 seconds apart, reset every check run. This is to make sure we never collect an event more than once. Since from and to are the same, they will return empty data the first check run. This logic is used when collecting deployments as well.